force-app/classes/test/MethodSpyTest.cls

```diff
@@ -43,26 +44,27 @@ private class MethodSpyTest {
   static void givenSpyConfiguredWithoutArguments_whenCalledWithoutArgument_spyIsCalled() {
     // Arrange
     final MethodSpy sut = new MethodSpy('methodName');
-    sut.whenCalledWith(new List<Object>()).thenReturn('test');
+    sut.whenCalledWith(Params.of()).thenReturn('test');
```

Comment:
question/suggestion: should we either introduce a `Params.empty()` or have a `whenCalled().thenReturn(...)` syntax to avoid boilerplate?

Reply:
I like renaming `Params.of()` with `Params.empty`.
I like the syntactic sugar `whenCalled` as well.
I'll do both

Reply:
While trying to implement it I think the `whenCalled` API is not easy to separate from the `returns` 
`whenCalled` means when Called with empty parameters then return, but as we do not handle method overload the spy could be reused for same methods with different parameter number.
`returns` does not care about the parameter and can be used as a syntactic sugar for methods without argument.
I think it can add a bit of cognitive complexity / learning curve because we need to understand this distinction and it is not clear at first sight.

I agree it is a bit verbose to use `whenCalledWith(Params.empty())` to drive the behavior when no argument are needed but it can already be done with `returns`